WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m

```diff
@@ -79,6 +80,43 @@ - (XCElementSnapshot *)fb_lastSnapshot
   return self.lastSnapshot;
 }
 
+- (XCElementSnapshot *)fb_lastSnapshotFromQuery
```

Comment:
why cannot we just use fb_lastSnapshot?

Reply:
Because fb_lastSnapshot actually resolves the query *for that element*, which then creates a new complete snapshot from the device, and filters it down to the element (-> slow, happens for each element).
(It seems the "lastSnapshot" member is not set for child elements of a query.) fb_lastSnapshotFromQuery() finds the root query, obtains the rootSnapshot tree from that query, then applies the element's query to each snapshot object to find it's corresponding snapshot.

Reply:
it would be useful to have this description in the method docstring